src/repo_policy_compliance_client.py

```diff
@@ -42,7 +42,7 @@ def get_one_time_token(self) -> str:
         Returns:
             The one-time token to be used in a single request of repo policy compliance check.
         """
-        url = urljoin(self.base_url, "one-time-token")
+        url = urljoin(str(self.base_url), "one-time-token")
```

Comment:
I guess here and elsewhere, what is the reason to needing to do type casts please?

Reply:
AnyHttpUrl in pydantic v1 was returning str but in v2 they return Url class and has to be type casted to use in a lot of places unfortunately.